js/main.js

```diff
@@ -16,17 +16,137 @@ if (previousEntries) {
+  r1c1P.textContent = profileData.fullName;
+
+  row1Column1.appendChild(r1c1P);
+  row1.appendChild(row1Column1);
+  divRoot.appendChild(row1);
+
+  var row2 = document.createElement('div');
+  row2.setAttribute('class', 'row');
+
+  var row2Column1 = document.createElement('div');
+  row2Column1.setAttribute('class', 'column-half');
+
+  var pfpDiv = document.createElement('div');
+  pfpDiv.setAttribute('class', 'pfp-preview');
+
+  var pfpImage = document.createElement('img');
+  pfpImage.setAttribute('src', profileData.avatarUrl);
+  pfpImage.setAttribute('class', 'pfp');
+
+  pfpDiv.appendChild(pfpImage);
+  row2Column1.appendChild(pfpDiv);
+  row2.appendChild(row2Column1);
+
+  var row2Column2 = document.createElement('div');
+  row2Column2.setAttribute('class', 'info column-half');
+
+  var noClassDiv = document.createElement('div');
+
+  var userNameInfo = document.createElement('div');
+  userNameInfo.setAttribute('class', 'user-profile-info');
+
+  var genericUserIcon = document.createElement('img');
+  genericUserIcon.setAttribute('src', 'images/person-icon.png');
+  genericUserIcon.setAttribute('class', 'icons');
+
+  var userName = document.createElement('p');
+  userName.textContent = profileData.username;
+
+  var userLocationInfo = document.createElement('div');
+  userLocationInfo.setAttribute('class', 'user-profile-info');
+
+  var genericLocationIcon = document.createElement('img');
+  genericLocationIcon.setAttribute('src', 'images/location-icon.png');
+  genericLocationIcon.setAttribute('class', 'icons');
+
+  var userLocation = document.createElement('p');
+  userLocation.textContent = profileData.location;
+
+  var userBiosInfo = document.createElement('div');
+  var biosP = document.createElement('p');
+  biosP.textContent = profileData.bio;
+
+  userNameInfo.appendChild(genericUserIcon);
+  userNameInfo.appendChild(userName);
+
+  userLocationInfo.appendChild(genericLocationIcon);
+  userLocationInfo.appendChild(userLocation);
+
+  userBiosInfo.appendChild(biosP);
+
+  noClassDiv.appendChild(userNameInfo);
+  noClassDiv.appendChild(userLocationInfo);
+  noClassDiv.appendChild(userBiosInfo);
+
+  row2Column2.appendChild(noClassDiv);
+  row2.appendChild(row2Column2);
+
+  divRoot.appendChild(row2);
+
+  return divRoot;
+}
+
+var allDataView = document.querySelectorAll('[data-view]');
+
+function viewSwapping(dataView) {
+  for (var i = 0; i < allDataView.length; i++) {
+    if (allDataView[i].getAttribute('data-view') !== dataView) {
+      allDataView[i].className = 'hide';
+    } else {
+      allDataView[i].className = 'show';
+
+      if (allDataView[i].getAttribute('data-view') === 'profile') {
+        if (allDataView[i].hasChildNodes()) {
+          var previousUserEntryPage = document.querySelector('#user-page');
+
+          allDataView[i].removeChild(previousUserEntryPage);
+        }
+
+        allDataView[i].appendChild(profilePage(data.entries[data.entries.length - 1]));
```

Comment:
You should just be passing in `data.profile` once you revise for my notes above 👍